### PR TITLE
DEV-21564 documented monitoring API

### PR DESCRIPTION
**Pull request by jeloba**
Documented the new monitoring API for 2019.10.

### CODE REVIEW

**Comment by hjohannsen** [comment]
Fine with me though I'm not always 100% sure about the English.

File: apiary.apib [hunk 1]
```diff
@@ -2081,6 +2081,86 @@ Retrieves active notifications from the platform.
 
 
 
+# Group Monitoring API
+
+The resources in this group provides information about the health and performace of the Acrolinx
```
Comment (ska2342) [inline apiary.apib]: subject verb agreement

File: apiary.apib [hunk 2]
```diff
@@ -2081,6 +2081,86 @@ Retrieves active notifications from the platform.
...
+platform. Accessing this information requires the permission "Access monitoring API." This privilege
```
Comment (ska2342) [inline apiary.apib]: Where?

File: apiary.apib [hunk 3]
```diff
@@ -2081,6 +2081,86 @@ Retrieves active notifications from the platform.
...
+* [general health](#monitoring-api-get-health-metrics)
+* [checking metrics](#monitoring-api-get-check-metrics)
+
+If you have the permission to access these values, the
+[metrics index](#monitoring-api-get-available-metrics) will provide links to these metrics.
+
+
+## Get Available Metrics [GET /api/v1/monitoring]
+
+Lists available metrics according to the permissions of the API user. If acces is granted a link to
```
Comment (ska2342) [inline apiary.apib]: typo access

File: apiary.apib [hunk 4]
```diff
@@ -2081,6 +2081,86 @@ Retrieves active notifications from the platform.
...
+
+|Key             | Target
+|----------------|-----------------
+|`checkMetrics`  |[check monitoring resource](#monitoring-api-get-check-metrics)
+|`healthMetrics` |[server health monitoring resource](#monitoring-api-get-health-metrics)
+
++ Response 200 (application/json)
+
+        {
+          "lins": {
```
Comment (ska2342) [inline apiary.apib]: lins?

File: apiary.apib [hunk 5]
```diff
@@ -2081,6 +2081,86 @@ Retrieves active notifications from the platform.
...
+          }
+        }
+
+## Get Check Metrics [GET /api/v1/monitoring/checks]
+
+Provides a summary of the check activity on the Acrolinx platform.
+
+|Field                    |Contents
+|-------------------------|-------------------
+|`receivedChecks`         |Total number of check request received by the plaform.
```
Comment (ska2342) [inline apiary.apib]: requests .... plural

File: apiary.apib (hunk 2 above)
Comment (jeloba) [reply apiary.apib]: Is it useful to link the docs page from here? https://docs.acrolinx.com/coreplatform/latest/en/user-management

File: apiary.apib (hunk 5 above)
Comment (ska2342) [reply apiary.apib]: Also inconsistent with following items. "Checks" vs "check requests"

File: apiary.apib [hunk 6]
```diff
@@ -2081,6 +2081,86 @@ Retrieves active notifications from the platform.
...
+|`successfulChecks`       |Total number of completed checks that did not result in an error.
+|`failedChecks`           |Total number of completed checks that resulted in an error.
+|`unparseableDocuments`   |Total number of checks that failed due to parsing errors.
+|`currentQueueLength`     |Current number of check request waiting for a free language server.
```
Comment (ska2342) [inline apiary.apib]: requests

File: apiary.apib [hunk 7]
```diff
@@ -2081,6 +2081,86 @@ Retrieves active notifications from the platform.
...
++ Response 200 (application/json)
+
+    + Attributes (object)
+        + data (object)
+            + receivedChecks: 54132 (number) - Total number of checks received
+            + successfulChecks: 54130 (number) - Total number of successful checks
+            + failedChecks (number): 2 - Total number of failed checks
+            + unparseableDocuments: 2 (number) - Total number of checks failed due to parsing errors
+            + currentQueueLength: 0 (number) - Current number of waiting checks
+            + totalQueueTime: 1563 (number) - Total time checks spend waiting
```
Comment (ska2342) [inline apiary.apib]: spent

File: apiary.apib [hunk 8]
```diff
@@ -2081,6 +2081,86 @@ Retrieves active notifications from the platform.
...
+ 
+
+## Get Health Metrics [GET /api/v1/monitoring/health]
+
+A general summary of the health of the Acrolinx platform.
+
+|Field                    |Contents
+|-------------------------|-------------------
+|`languagesReady`         |A hash containing all configured languages as keys. If Acrolinx is ready to check in a language ist value will be true.
```
Comment (ska2342) [inline apiary.apib]: its

**Comment by jeloba** [comment]
Apparently I "saved" the sidebar and not the document after I checked/corrected it with Acrolinx.

File: apiary.apib (hunk 2 above)
Comment (ska2342) [reply apiary.apib]: It's sufficient to just say that the permission is required on the Acrolinx Platform or must be granted in the dashboard.

File: apiary.apib [hunk 9]
```diff
@@ -2083,11 +2083,10 @@ Retrieves active notifications from the platform.
...
 # Group Monitoring API
 
-The resources in this group provides information about the health and performace of the Acrolinx
-platform. Accessing this information requires the permission "Access monitoring API." This privilege
-is not granted to all users. Make sure the user associated with your API token has the requisite roles.
+The resources in this group provide information about the health and performance of the Acrolinx Platform. Accessing this information requires the permission "Access monitoring API." This privilege
+isn’t granted to all users. Make sure the user associated with your API token has the requisite roles.
 
-Monitoring data is divided for the folloing topics:
+Monitoring data is divided for the following topics:
```
Comment (ska2342) [inline apiary.apib]: divided? provided?

Comment (ska2342) [reply apiary.apib]: Even better: remove those paragraphs because they duplicate information coming next.

**Comment by ska2342** [comment]
Please also mention somewhere in the introduction which Platform version introduced the monitoring API.

File: apiary.apib [hunk 10]
```diff
@@ -2081,6 +2081,82 @@ Retrieves active notifications from the platform.
...
 
+# Group Monitoring API
+
+The resources in this group provide information about the health and performance of the Acrolinx Platform.
+Accessing this information requires the permission "Access monitoring API." This privilege isn’t granted
+to all users. Make sure the user associated with your API token has the requisite roles.
+
+For a quick check of your permissions see if the [metrics index](#monitoring-api-get-available-metrics) 
+contains links. More information about permission management can be found in the
+[Acrolinx documentation](https://docs.acrolinx.com/coreplatform/latest/en/user-management).
```
Comment (jeloba) [inline apiary.apib]: @ska2342 How does this sound?

**Pull request merged**